### PR TITLE
2 small CQ optimizations.

### PR DESCRIPTION
**Pull request by sandreenko**
Found during `JitDoOldStructRetyping` work.

Diffs are positive:
```
Crossgen CodeSize Diffs for System.Private.CoreLib.dll, framework assemblies for  default jit
Summary of Code Size diffs:
(Lower is better)
Total bytes of diff: -31060 (-0.10% of base)
3169 total methods with Code Size differences (3167 improved, 2 regressed)

PMI CodeSize Diffs for System.Private.CoreLib.dll, framework assemblies for  default jit
Summary of Code Size diffs:
(Lower is better)
Total bytes of diff: -40121 (-0.08% of base)
4964 total methods with Code Size differences (4952 improved, 12 regressed)
```

Need these changes to get rid of some regressions with `!JitDoOldStructRetyping`.

A simple diff example, `-38 (-35.85% of base) : System.Private.CoreLib.dasm - Math:Truncate(Decimal):Decimal`:
before
```
IN0001: 000008 mov      ecx, dword ptr [rdx]
IN0002: 00000A mov      dword ptr [V05 rsp+20H], ecx
IN0003: 00000E mov      ecx, dword ptr [rdx+4]
IN0004: 000011 mov      dword ptr [V06 rsp+24H], ecx
IN0005: 000015 mov      ecx, dword ptr [rdx+8]
IN0006: 000018 mov      dword ptr [V07 rsp+28H], ecx
IN0007: 00001C mov      edx, dword ptr [rdx+12]
IN0008: 00001F mov      dword ptr [V08 rsp+2CH], edx
IN0009: 000023 mov      edx, dword ptr [V05 rsp+20H]
```
after:
```
IN0001: 000008 movups   xmm0, xmmword ptr [rdx]
IN0002: 00000B movups   xmmword ptr [V03 rsp+20H], xmm0
IN0003: 000010 mov      edx, dword ptr [V05 rsp+20H]
```

### CODE REVIEW

File: src/coreclr/src/jit/morph.cpp [hunk 1]
```diff
@@ -10600,8 +10600,20 @@ GenTree* Compiler::fgMorphCopyBlock(GenTree* tree)
         // If we passed the above checks, then we will check these two
         if (!requiresCopyBlock)
         {
+            // It is not always profitable to do field by field init for structs that are allocated to memory.
```
Comment (sandreenko) [inline src/coreclr/src/jit/morph.cpp:10603]: found after Carol's similar change for block init #36146

**Comment by sandreenko** [comment]
PTAL @CarolEidt @dotnet/jit-contrib

**Review by CarolEidt — APPROVED** [review]
LGTM - thanks! and thanks for the detailed comments

**Comment by sandreenko** [comment]
The failures are known unrelated issues,

**Pull request merged**